lib/common/options.c

```diff
@@ -656,32 +656,48 @@ pcmk__format_option_metadata(const char *name, const char *desc_short,
     g_string_append(s, "  <parameters>\n");
 
     for (lpc = 0; lpc < len; lpc++) {
-        const char *long_desc = option_list[lpc].description_long;
+        const char *opt_name = option_list[lpc].name;
+        const char *opt_type = option_list[lpc].type;
+        const char *opt_values = option_list[lpc].values;
+        const char *opt_default = option_list[lpc].default_value;
+        const char *opt_desc_short = option_list[lpc].description_short;
+        const char *opt_desc_long = option_list[lpc].description_long;
+
+        // The standard requires long and short parameter descriptions
+        if ((opt_desc_short == NULL) && (opt_desc_long == NULL)) {
+            continue;
+        } else if (opt_desc_short == NULL) {
+            opt_desc_short = opt_desc_long;
+        } else if (opt_desc_long == NULL) {
+            opt_desc_long = opt_desc_short;
+        }
 
-        if (long_desc == NULL) {
-            long_desc = option_list[lpc].description_short;
-            if (long_desc == NULL) {
-                continue; // The standard requires a parameter description
-            }
+        // The standard requires a parameter type
+        if (opt_type == NULL) {
+            continue;
```

Comment:
Since this is an internal function, this would be a bug, so maybe just assert that the \*desc\* and type fields are non-null so we're forced to come up with something reasonable rather than silently ignore a malformed entry

Though in that case, we should add regression tests for all callers so problems don't slip through

Reply:
The Makefile already runs `daemons/<daemon>/pacemaker-$daemon metadata > pacemaker-$daemon.xml` for each daemon that supports it, via `mk/man.mk`. So that makes sure we don't assert. If we want, we can add regression tests to `cts-cli` that actually check the output. I'll do that since it should be easy, and we can decide whether to keep it.